unit-tests/src/test/scala/java/nio/file/FilesSuite.scala

```diff
@@ -891,6 +894,64 @@ object FilesSuite extends tests.Suite {
     }
   }
 
+  // This test was inspired by Issue #1354
+  // "NIO File Walker fails on broken links."
+  test("Files.walkFileTree respects FileVisitOption.FOLLOW_LINKS") {
+    withTemporaryDirectory { dirFile =>
+      val dir = dirFile.toPath()
+      val f0  = dir.resolve("f0")
+      val f1  = dir.resolve("f1")
+
+      val brokenLink    = dir.resolve("brokenlink")
+      val missingTarget = dir.resolve("missingtarget")
+
+      Files.createFile(f0)
+      Files.createFile(f1)
+
+      assert(Files.exists(f0) && Files.isRegularFile(f0))
+      assert(Files.exists(f1) && Files.isRegularFile(f1))
+
+      Files.createFile(missingTarget)
+
+      assert(
+        Files.exists(missingTarget) &&
+          Files.isRegularFile(missingTarget))
+
+      // Create valid symbolic link from brokenLink to missingTarget,
+      // then remove missingTarget to break link.
+
+      Files.createSymbolicLink(brokenLink, missingTarget)
+
+      assert(Files.exists(brokenLink) && Files.isSymbolicLink(brokenLink),
+             "File brokenLink does not exist or is not a symbolic link.")
+
+      Files.delete(missingTarget)
+
+      assert(!Files.exists(missingTarget),
+             "File missingTarget should not exist.")
+
+      val visitor = new QueueingVisitor()
+
+      // will not follow links, so broken link does not throw exception.
+      Files.walkFileTree(dir, visitor)
+
+      val result   = visitor.length
+      val expected = 4
+
+      assert(result == expected, s"result: $result != expected: $expected")
+
+      // Now follow the broken link and throw exception.
+      val visitor_2 = new QueueingVisitor()
```

Comment:
Not a fan of `_` in variable names in scala.

Reply:
The use of the `_2` also implies that perhaps this should be a separate test. It would make debugging easier if there is a future regression.

Reply:
re: underscore.  
    I can change that to conform to Scala practice, or, better yet the entire variable name.
I find that my poor eyes miss the terminal digit in `visitor2`.  

re: separate test. 

    I will do that. I'll probably create a method to do the setup for the two tests in common, then get down
to the intent of the test:  Isolate the setup.

Reply:
I agree that we should split it into 2 or more tests. Underscores in variable names are also generally frowned upon.

Reply:
I followed the sense of the meeting and split the test into two.  I have each of the
tests calling a common setup method to keep them in synch. 

I do have to admit that the new code looks better, at least to me.

Also, here are now no  underscores in variable names.  I will avoid them in Scala in 
the future.